### PR TITLE
Bootstrap structure for a widget examples repo

### PR DESCRIPTION
**Pull request by jtpio**
First step towards a "Widget Examples" repository for `ipywidgets`: https://github.com/jupyter-widgets/ipywidgets/issues/3136

Heavily inspired by the JupyterLab Extension Examples: https://github.com/jupyterlab/extension-examples

The idea is to set up the basic structure for multiple custom widget examples in the same repo, so that we have the tools and CI in place to iterate and easily add new examples.

## TODO

- [x] Set up the structure and tooling
- [x] Add dev instructions to work on the repo 
- [x] Add `hello-world` example bootstrapped from the TS cookiecutter: https://github.com/jupyter-widgets/widget-ts-cookiecutter
- [x] Make sure CI passes

### CODE REVIEW

**Comment by jtpio** [comment]
Example CI run in the fork: https://github.com/jtpio/custom-ipywidget-howto/pull/1

**Comment by jtpio** [comment]
Seeing some `ESOCKETTIMEDOUT` on `macos-latest`:

https://github.com/jtpio/custom-ipywidget-howto/pull/1/checks?check_run_id=2120510605

Maybe we can keep `ubuntu-latest` on CI for now and add back `macos` and `windows` later on.

**Comment by jtpio** [comment]
Listing a couple of items to iterate on in follow-up PRs:

- Enable tests on CI with `pytest`
- Test on `macos` and `windows` on CI
- Add the existing two examples `jslink-custom-widgets` and `connecting-widgets` to the list of examples

**Comment by jtpio** [comment]
cc @ianhi if you want to have a look.

This PR is mostly about adding the tooling and the structure, more than actually working on the examples themselves.

Maybe there could be a couple more iterations before renaming the repo to `widget-examples` and moving it to the `jupyter-widgets` organization on GitHub.

**Comment by ianhi** [comment]
Hi @jtpio thanks so much for starting on this! I've been busy at the start of this week but will look at this tomorrow (today now?) evening.

**Comment by ianhi** [comment]
I had been playing a bit fast and loose with some of the formatting which prettier then mangled. So I updated that and removed some outdated portions of the readme.

File: connecting-widgets/README.md [hunk 1]
```diff
@@ -9,14 +8,12 @@ Exploring how to give access of a widget's typescript model to a different widge
 
 **Important questions:**
 
-Q: Is this good practice?  - A: I have no idea
+Q: Is this good practice? - A: I have no idea
```
Comment (ianhi) [inline connecting-widgets/README.md:11]: @jtpio do you know the answer to this?

File: hello-world/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# Hello World
+
+Hello World Custom Widget Example.
+
+## Structure
+
+This example was bootstrapped from the [Jupyter TypeScript Cookiecutter](https://github.com/jupyter-widgets/widget-ts-cookiecutter) and follows the same structure.
```
Comment (ianhi) [inline hello-world/README.md:7]:
```suggestion
This example was bootstrapped from the [Jupyter Widget TypeScript Cookiecutter](https://github.com/jupyter-widgets/widget-ts-cookiecutter) and follows the same structure.
```

Is that a more correct phrasing?

**Review by ianhi — APPROVED** [review]
This is fantastic. Thanks so much for this. I have some follow up questions about how to best improve this but they can go into new issues.

**Pull request merged**